jupyter_server_fileid/extension.py

```diff
@@ -10,28 +11,33 @@ class FileIdExtension(ExtensionApp):
 
     file_id_manager_class = Type(
         klass=BaseFileIdManager,
-        help="""File ID manager instance to use.
+        help="""File ID manager class to use.
 
         Defaults to ArbitraryFileIdManager.
         """,
         config=True,
         default_value=ArbitraryFileIdManager,
     )
 
+    file_id_manager = Instance(
+        klass=BaseFileIdManager, help="An instance of the File ID manager.", allow_none=True
+    )
```

Comment:
Any reason for defining this instance attribute as a class trait instead of just declaring its type? I thought we only attributes as traits when we want them to be configurable.

```suggestion
    file_id_manager: BaseFileIdManager
```

I'm fine either way, but I find the type declaration easier to read because of its brevity.

Reply:
Well, traitlets goes a step further than type _hinting_. It validates the type too (similar to Pydantic). 

Traitlets is more of a type validation library; config is just a bonus feature. I tend to use traitlets to guard against attributes being set to improper types at runtime.

It's not strictly necessary here, since we set this instance ourselves. Because it's Python, though, it's technically possible to dynamically change this attribute at runtime and give it the wrong type. (Though I admit, the risk is low.)

Reply:
I'm also fine either way too. Let me know what you feel more comfortable doing. 

I agree; traitlets is verbose and the gain here might be small.

Reply:
Well, let's keep it. I'd love to explore the idea of a more concise implementation of `traitlets` someday, but alas, we're all short on time it seems.

Reply:
100%. You're not the only one feeling this sentiment 😅 A few of us have tried to create traitlet replacements, and not gained much momentum... (see [schematypes](https://github.com/deathbeds/schematypes), [msgspec](https://github.com/jcrist/msgspec), [cattrs](https://catt.rs/en/latest/), pydantic)

I think something like Pydantic, + config + logging + observability would be the shortest path to a traitlets replacement.